src/lib/normalize-url.js

```diff
@@ -0,0 +1,40 @@
+
+// Mobile apps give us full URL with domain and protocol.
+// Web apps could also initilize with a full URL
+// but they usually don't, and we the use the 
+// domain given by the stack through a DOM node
+// in the HTML page (depite it not being a full URL).
+//
+// This function normalize these different cases
+// and give a full URL object in response.
+// 
+// The `secure` parameter, if given, always has the
+// priority over the one from the URL.
+// We use a secure protocol by default if we can't decide.
+
```

Comment:
JSDoc style comments are picked up by automatic documentation tools are the norm in our code.

Example

```
  /**
   * Creates a {@link DocumentCollection} instance.
   *
   * @param  {String} doctype The collection doctype.
   * @return {DocumentCollection}
   */
```